### PR TITLE
Allow configuring a TextMapPropagator for SQLCommenter

### PR DESCRIPTION
**Pull request by simonbos**
This allows users to not rely on the global TextMapPropagator. It might be enough to close https://github.com/XSAM/otelsql/issues/192 because then users are free to implement a custom propagator which is specific for the SQLCommenter use case without polluting the global TextMapPropagator.

### CODE REVIEW

**Comment by XSAM** [comment]
Hi @simonbos, thanks for creating this PR.

I am not sure it can address #192 as it seems to require dynamic key-value based on query. The propagator can only propagate static values without the support of special context tools provided to users.

But it is useful to add an API for user to replacement the propagator used for sql commenter (related PR https://github.com/open-telemetry/semantic-conventions/pull/2495).

Could you add tests for cases to cover the new options with sql commenter enabled ? It probably can catch the nil pointer issue when this new option is not setup.

Please also update the changelog file.

File: option.go [hunk 1]
```diff
@@ -95,6 +96,14 @@ func WithSQLCommenter(enabled bool) Option {
 	})
 }
 
+// WithTextMapPropagator specifies a text map propagator to used by the SQLCommentor
+// option. If none is specified, the global text map propagator is used.
```
Comment (XSAM) [inline option.go]:
```suggestion
// WithTextMapPropagator specifies a text map propagator to used by the SQLCommenter
// option. If none is specified, the global text map propagator is used.
//
// Notice: This option is EXPERIMENTAL and may be changed or removed in a
// later release.
```

Since it is an option tied to the SQL SQLCommenter, it should be marked as EXPERIMENTAL.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Added
+
+- `WithTextMapPropagator` allows customization of the OTel text map propagator used by SQLCommenter. (#540)
+
```
Comment (XSAM) [inline CHANGELOG.md]:
```suggestion

  This is an experimental feature and may be changed or removed in a later release.
```

**Review by XSAM — COMMENTED** [review]
> I added an extra test case for the nil case to the commenter_test, any other test case you would like to see?

Could you add a test case for the new added option in `TestOptions` from `option_test.go`?

Regarding the nil issue, I could find the issue anymore. It could be my misread.

**Comment by simonbos** [comment]
Sorry for the delay, I added the test case and added the experimental clarification.

**Review by XSAM — APPROVED** [review]
@simonbos thanks for your contribution!
